contracts/validators/CredentialAtomicQueryV3Validator.sol

```diff
@@ -179,9 +179,9 @@ contract CredentialAtomicQueryV3Validator is CredentialAtomicQueryValidator {
         require(nullifierSessionID == 0 || nullifier != 0, "Invalid nullify pub signal");
     }
 
-    function _checkAuth(uint256 challenge) internal view {
+    function _checkAuth(uint256 challenge, address ethIdentityOwner) internal view {
         require(
-            PrimitiveTypeUtils.int256ToAddress(challenge) == tx.origin,
+            PrimitiveTypeUtils.int256ToAddress(challenge) == ethIdentityOwner,
```

Comment:
we need to extract user id not from the challenge, but from userID public signal, and convert ethIdentityOwner to id